frontend/src/app/services/auth-page-guard/auth-page-guard.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot} from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { RouteName } from 'src/app/enums/route-name-enun';
+import { AuthService } from '../auth/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthPageGuardService implements CanActivate {
+
+  constructor(
+    private router: Router,
+    private authService: AuthService, 
+  ) { }
+
+  public canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    if(!this.authService.isAuthenticated()) {
```

Comment:
It is very strange logic, If to read this logic, if you are not authenticated this guard return true and give you access, if you are authenticated guard navigate you auth page!

How is it pissible?